src/csharp/ext/grpc_csharp_ext.c

```diff
@@ -927,7 +927,8 @@ GPR_EXPORT void GPR_CALLTYPE grpcsharp_metadata_credentials_notify_from_plugin(
 }
 
 typedef void(GPR_CALLTYPE *grpcsharp_metadata_interceptor_func)(
-  void *state, const char *service_url, grpc_credentials_plugin_metadata_cb cb,
+  void *state, const char *service_url, const char *method_name,
```

Comment:
any reason why you're not passing the whole context here (as opposed to individual args)? That will make things easier when you add the auth_context.